frontend/packages/aquascope-editor/src/editor-utils/stepper.tsx

```diff
@@ -193,9 +181,67 @@ let stepLocation = (step: PermissionsStateStep): number => {
   return step.location.char_end;
 };
 
+let StepTable = ({ rows }: { rows: [string, PermissionsDataDiff][] }) => {
+  return (
+    <table className="perm-step-table">
+      {rows.map(([path, diffs], i: number) => (
+        <PermDiffRow key={i} path={path} diffs={diffs} />
+      ))}
+    </table>
+  );
+};
+
+let StepTableWidget = ({
+  spaces,
+  focused,
+  hidden,
+  init,
+}: {
+  spaces: string;
+  focused: [string, PermissionsDataDiff][];
+  hidden: [string, PermissionsDataDiff][];
+  init: boolean;
+}): JSX.Element => {
+  let [display, setDisplay] = useState(init);
+  let [displayAll, setDisplayAll] = useState(false);
+  let arrowOut = "»";
+  let arrowIn = "«";
+
+  let hiddenDropdown =
+    hidden.length > 0 ? (
+      <>
+        <div className="step-table-dropdown">
+          <i className={displayAll ? "fa fa-minus" : "fa fa-ellipsis-h"} />
+        </div>
+        <div className={displayAll ? "" : " hidden-height"}>
+          <StepTable rows={hidden} />
+        </div>
+      </>
+    ) : null;
+
+  return (
+    <div className="perm-step-widget">
+      <span onClick={() => setDisplay(!display)}>
+        {display ? arrowIn : arrowOut}
+      </span>
+
+      <div
+        className={"step-widget-container" + (display ? "" : " hidden-width")}
```

Comment:
Btw if you find yourself dealing with conditional class names a lot, this is handy utility: https://www.npmjs.com/package/classnames

e.g. you can do

```jsx
<div className={classNames("step-widget-container, {"hidden-width": display})} />
```